app/ide-desktop/lib/client/watch.ts

```diff
@@ -109,18 +109,32 @@ process.on('SIGINT', () => {
     })
 })
 
-while (true) {
+/**
+ * Starts the electron process with the IDE.
+ */
+function startElectronProcess() {
```

Comment:
I refactored the watch script and disabled continuous restarting of the app because it leads to a worse experience(e.g. when I close the app it restarts immediately. Same happens when process fails and leads to infinite cycle of restarting app)

Reply:
yeah makes sense. i kept it around because it was a thing in the old watch script, but as both dashboard and GUI use vite, HMR should more or less take care of everything